core/templates/pages/story-editor-page/editor-tab/story-node-editor.directive.ts

```diff
@@ -123,9 +123,20 @@ angular.module('oppia').directive('storyNodeEditor', [
             if (newTitle === $scope.currentTitle) {
               return;
             }
-            StoryUpdateService.setStoryNodeTitle(
-              $scope.story, $scope.getId(), newTitle);
-            $scope.currentTitle = newTitle;
+            var titleIsValid = true;
+            for (var idx in $scope.story.getStoryContents().getNodes()) {
+              var node = $scope.story.getStoryContents().getNodes()[idx];
+              if (node.getTitle() === newTitle) {
+                titleIsValid = false;
+                AlertsService.addInfoMessage(
+                  'A chapter already exists with given title.', 5000);
```

Comment:
Do we have this check in backend?

Reply:
writing here so I can track this.

Reply:
Oh, no, it isn't there, will add it in #9056

Reply:
Do we want that to be a part of the current release? (Considering this can damage production data.)

Reply:
Done

Reply:
Thanks for verification, should we release this in the current release so that no one adds duplicate names in this month?